File: jeongho/leetcode/2.Best Time to Buy and Sell Stock.java
import java.util.ArrayList;
import java.util.Arrays;

class Solution {
    public static void main(String[] args) {
        Solution solution = new Solution();
        int print = solution.maxProfit(new int[] { 7, 1, 5, 3, 6, 4 });

        System.out.println(print);
    }

    public int maxProfit(int[] prices) {
        int minInt = Integer.MAX_VALUE;
        int maxInt = 0;

        for (int i = 0; i < prices.length; i++) {
            if (prices[i] < minInt) {
                minInt = prices[i];
            } else if (prices[i] - minInt > maxInt) {
                maxInt = prices[i] - minInt;
            }
        }
        return maxInt;
    }
}

File: wockd9600/leetcode/array/2. Best Time to Buy and Sell Stock.py
def maxProfit(prices) -> int:
        lm = 10001
        m = 0
        for i in range(len(prices)):
            if prices[i] < lm: lm = prices[i]
            if prices[i] - lm > m: m = prices[i] - lm
        return m

print(maxProfit([7,1,5,3,6,4]))